src/sentry/integrations/bitbucket/search.py

```diff
@@ -37,32 +40,39 @@ def installation_class(self):
         return BitbucketIntegration
 
     def handle_search_issues(self, installation: T, query: str, repo: str | None) -> Response:
-        assert repo
+        with self.record_event(
+            SourceCodeSearchEndpointInteractionType.HANDLE_SEARCH_ISSUES
+        ).capture() as lifecycle:
+            assert repo
 
-        full_query = f'title~"{query}"'
-        try:
-            response = installation.search_issues(query=full_query, repo=repo)
-        except ApiError as e:
-            if "no issue tracker" in str(e):
-                logger.info(
-                    "bitbucket.issue-search-no-issue-tracker",
-                    extra={"installation_id": installation.model.id, "repo": repo},
-                )
-                return Response(
-                    {"detail": "Bitbucket Repository has no issue tracker."}, status=400
-                )
-            raise
+            full_query = f'title~"{query}"'
+            try:
+                response = installation.search_issues(query=full_query, repo=repo)
+            except ApiError as e:
+                if "no issue tracker" in str(e):
+                    lifecycle.record_halt()
```

Comment:
We've updated `record_halt` and `record_failure` to take message strings when we don't have an exception to pass so that our logs are a bit more informative + we can run them through log analytics to get counts of different halt types. I think it'd be helpful to track these on most of the halts in this PR.

Raj has also been adding enum strings for shared halt reasons to keep the naming consistent between call sites as well.